### PR TITLE
feature: Added transform with monitoring pipeline step in transformer

### PR DESCRIPTION
**Pull request by keshav-chandak**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/transformer.py [hunk 1]
```diff
@@ -247,6 +250,155 @@ def transform(
...
+        supplied_baseline_statistics: str = None,
+        supplied_baseline_constraints: str = None,
+        wait: bool = True,
+        pipeline_name: str = None,
+        role: str = None,
+    ):
+        """Runs a transform job with monitoring job.
+
+        Note that this function will not start a transform job immediately,
+        instead, it will create a SageMaker Pipeline and execute it.g331
```
Comment (adeel-zafar) [inline src/sagemaker/transformer.py:277]: Typo, `execute it.g331`

**Review by adeel-zafar — CHANGES_REQUESTED** [review]

File: src/sagemaker/transformer.py (hunk 1 above)
Comment (keshav-chandak) [reply src/sagemaker/transformer.py]: fixed

**Review by adeel-zafar — APPROVED** [review]
LGTM

**Review by jerrypeng7773 — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests

**Review by knikure — APPROVED** [review]

File: src/sagemaker/transformer.py [hunk 2]
```diff
@@ -266,6 +269,155 @@ def transform(
...
+            monitor_before_transform=monitor_before_transform,
+            supplied_baseline_constraints=supplied_baseline_constraints,
+            supplied_baseline_statistics=supplied_baseline_statistics,
+        )
+
+        pipeline_name = (
+            pipeline_name if pipeline_name else f"TransformWithMonitoring{int(time.time())}"
+        )
+        # if pipeline exists, just start the execution
+        from sagemaker.workflow.pipeline import Pipeline
```
Comment (rahven14) [inline src/sagemaker/transformer.py:407]: Move import to top of the file

File: tests/integ/test_transformer.py [hunk 3]
```diff
@@ -645,3 +646,66 @@ def _create_transformer_and_transform_job(
...
+        sagemaker_session=sagemaker_session,
+        entry_point=os.path.join(os.path.join(DATA_DIR, "xgboost_abalone"), "inference.py"),
+        enable_network_isolation=True,
+    )
+
+    xgb_model.deploy(_INSTANCE_COUNT, _INSTANCE_TYPE)
+
+    transform_output = f"s3://{sagemaker_session.default_bucket()}/{pipeline_name}Transform"
+    transformer = Transformer(
+        model_name=xgb_model.name,
```
Comment (rahven14) [inline tests/integ/test_transformer.py:683]: this line has the potential to throw an exception, if xgb_model.name is not accessible. So please have a check like,
model_name = xgb_model.name if xgb_model else None or something to fill in the else case.

File: tests/integ/test_transformer.py [hunk 4]
```diff
@@ -645,3 +646,66 @@ def _create_transformer_and_transform_job(
...
+    xgb_model = XGBoostModel(
+        model_data=xgb_model_data_s3,
+        framework_version="1.3-1",
+        role=role,
+        sagemaker_session=sagemaker_session,
+        entry_point=os.path.join(os.path.join(DATA_DIR, "xgboost_abalone"), "inference.py"),
+        enable_network_isolation=True,
+    )
+
+    xgb_model.deploy(_INSTANCE_COUNT, _INSTANCE_TYPE)
```
Comment (rahven14) [inline tests/integ/test_transformer.py:679]: are they private variables defined somewhere, not able to find it.

**Pull request merged**